test/blackbox/api/dds-pim/PubSubWriter.hpp

```diff
@@ -514,13 +514,25 @@ class PubSubWriter
 
     void send(
             std::list<type>& msgs,
-            uint32_t milliseconds = 0)
+            uint32_t milliseconds = 0,
+            bool send_with_params = false)
```

Comment:
1. Instead of having a boolean and an attribute, I would directly have a `fastdds::rtps::WriteParams* params = nullptr`
2. Might be nice to have similar changes in the two `send_sample` methods below

Reply:
Since we dont have API for writing with both an InstanceHandle and WriteParams, I did the changes only in the first `send_sample` methods below